admin-base/ui.apps/src/main/js/perAdminApp.js

```diff
@@ -301,97 +301,111 @@ function processLoaders(loaders) {
  *
  */
 function loadContentImpl(initialPath, firstTime, fromPopState) {
-    logger.fine('loading content for', initialPath)
-    view.admin.consoleErrors = false
-
-    if(!runBeforeStateActions() ) {
-        logger.fine('not allowed to switch state')
-        return
-    }
-
-    var pathInfo = makePathInfo(initialPath.toString())
-    var path = pathInfo.path
-
-    var dataUrl = pagePathToDataPath(path)
-
-    view.status = undefined;
-
-    api.populateUser()
+  logger.fine('loading content for', initialPath);
+  view.admin.consoleErrors = false;
+
+  if (!runBeforeStateActions()) {
+    logger.fine('not allowed to switch state');
+    return;
+  }
+
+  var pathInfo = makePathInfo(initialPath.toString());
+  var path = pathInfo.path;
+
+  var dataUrl = pagePathToDataPath(path);
+
+  view.status = undefined;
+
+  return api
+    .populateUser()
+    .then(function() {
+      if (pathInfo.suffixParams.path) {
+        const segments = pathInfo.suffixParams.path.split('/');
+        if (segments.length >= 3) {
+          if (view.state.tenant && view.state.tenant.name !== segments[2]) {
+            logger.error('tenant mismatch');
+          } else {
+            logger.fine('tenant missing');
+            return api.populateTenants().then(() => {
+              const tenant = view.admin.tenants.filter(
+                (node) => node.name === segments[2]
+              );
+              if (tenant.length >= 1) {
+                view.state.tenant = tenant[0];
+              } else {
+                logger.error('tenant does not exist or no access to tenant');
+              }
+            });
+          }
+        }
+      }
+    })
+    .then(function() {
+      return api
```

Comment:
Is the value returned from this anonymous function used? Or is this a prettier idea?